lib/web/sign.go

```diff
@@ -0,0 +1,139 @@
+	"github.com/julienschmidt/httprouter"
+	"golang.org/x/exp/slices"
+
+	apidefaults "github.com/gravitational/teleport/api/defaults"
+	"github.com/gravitational/teleport/lib/client/identityfile"
+	"github.com/gravitational/teleport/lib/httplib"
+	"github.com/gravitational/teleport/lib/reversetunnel"
+	"github.com/gravitational/teleport/lib/srv"
+	"github.com/gravitational/teleport/lib/utils"
+)
+
+/* signCertKeyPair returns the necessary files to set up mTLS for other services
+This is the equivalent of running the tctl command
+As an example, requesting:
+POST /webapi/sites/mycluster/sign
+{
+	"hostname": "pg.example.com",
+	"ttl": "2190h",
+	"format": "db"
+}
+
+Should be equivalent to running:
+   tctl auth sign --host=pg.example.com --ttl=2190h --format=db
+
+This endpoint returns a tar.gz compressed archive containing the required files to setup mTLS for the service.
+*/
+func (h *Handler) signCertKeyPair(w http.ResponseWriter, r *http.Request, p httprouter.Params, site reversetunnel.RemoteSite) (interface{}, error) {
+	req := &signCertKeyPairReq{}
+	if err := httplib.ReadJSON(r, &req); err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	if err := req.CheckAndSetDefaults(); err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	virtualFS := identityfile.NewInMemoryConfigWriter()
+
+	mTLSReq := srv.GenerateMTLSFilesRequest{
+		ClusterAPI:          h.auth.proxyClient,
+		Principals:          []string{req.Hostname},
+		OutputFormat:        req.Format,
+		OutputCanOverwrite:  true,
+		OutputLocation:      "server",
+		IdentityFileWriter:  virtualFS,
+		TTL:                 req.TTL,
+		HelperMessageWriter: nil,
+	}
+	filesWritten, err := srv.GenerateMTLSFiles(r.Context(), mTLSReq)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	archiveName := fmt.Sprintf("teleport_mTLS_%s.tar.gz", req.Hostname)
+
+	// https://www.postgresql.org/docs/current/libpq-ssl.html
+	// On Unix systems, the permissions on the private key file must disallow any access to world or group;
+	//  achieve this by a command such as chmod 0600 ~/.postgresql/postgresql.key.
+	// Alternatively, the file can be owned by root and have group read access (that is, 0640 permissions).
+	fileMode := fs.FileMode(teleport.FileMaskOwnerOnly) // 0600
+	archiveBytes, err := utils.CompressTarGzArchive(filesWritten, virtualFS, fileMode)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	// Set file name
+	w.Header().Set("Content-Disposition", fmt.Sprintf(`attachment;filename="%v"`, archiveName))
+
+	// ServeContent sets the correct headers: Content-Type, Content-Length and Accept-Ranges.
+	// It also handles the Range negotiation
+	http.ServeContent(w, r, archiveName, time.Now(), bytes.NewReader(archiveBytes.Bytes()))
+
+	return nil, nil
+}
+
+type signCertKeyPairReq struct {
+	Hostname  string `json:"hostname,omitempty"`
+	FormatRaw string `json:"format,omitempty"`
+	TTLRaw    string `json:"ttl,omitempty"`
+	Format    identityfile.Format
+	TTL       time.Duration
```

Comment:
If these aren't supposed to be marshaled/unmarshaled, add explicit json tag:
```suggestion
	Format    identityfile.Format `json:"-"`
	TTL       time.Duration `json:"-"`
```